src/extensions/Statiq.ApiClient/ReadApiWithClient.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+using Statiq.Common;
+
+namespace Statiq.ApiClient
+{
+    /// <summary>
+    /// Outputs metadata for information from any API with its client.
+    /// </summary>
+    /// <remarks>
+    /// This modules used to submit requests to the API with the API client. Because of the possible large
+    /// number of different kinds of requests, this module does not attempt to provide a fully abstract wrapper
+    /// around the API. Instead, it simplifies the housekeeping involved in setting up an
+    /// API client and requires you to provide functions that fetch whatever data you need. Each request
+    /// will be sent for each input document.
+    /// </remarks>
+    /// <category>Metadata</category>
+    public class ReadApiWithClient<TClient> : ParallelSyncModule
+        where TClient : class
+    {
+        private readonly Dictionary<string, Func<IDocument, IExecutionContext, TClient, object>> _requests
+            = new Dictionary<string, Func<IDocument, IExecutionContext, TClient, object>>();
+
+        private readonly TClient _client;
+
+        private Action<IDocument, IExecutionContext, TClient> _init;
+
+        private string _clientName;
+
+        /// <summary>
+        /// Creates a connection to the API using client.
+        /// </summary>
+        /// <param name="client">The API client to use.</param>
+        public ReadApiWithClient(TClient client)
+        {
+            _client = client ?? throw new ArgumentException("Argument is null", nameof(client));
+        }
+
+        /// <summary>
+        /// Initialize the API client.
+        /// </summary>
+        /// <param name="init">An API client initialization action.</param>
+        /// <returns>The current module instance.</returns>
+        public ReadApiWithClient<TClient> WithClientInitialization(Action<TClient> init)
+        {
+            if (init == null)
+            {
+                throw new ArgumentException("Argument is null or empty", nameof(init));
```

Comment:
Switch the argument null checks to `init.ThrowIfNull(nameof(init))` - I've adopted that convention across the board recently.

Reply:
It will be done.